strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java

```diff
@@ -618,4 +624,57 @@ private void modifyInLock(final Consumer<MutableConfiguration> operation,
         }
     }
 
+    @Override
+    public boolean setMaxUploadFileSize(String maxUploadSizeString) throws SizeFormatValidationException
+    {
+        if (!(this.multipartConfigElement instanceof UpdatableMultipartConfigElement))
+        {
+            return false;
+        }
+
+        maxUploadSizeString.replaceAll("\\s+","");
+        maxUploadSizeString = maxUploadSizeString.toUpperCase();
+
+        Long uploadSize;
+        if(maxUploadSizeString.equals("UNLIMITED"))
+        {
+            uploadSize = -1l;
+            maxUploadSizeString = maxUploadSizeString.toLowerCase();
+        }
+        else
+        {
+            uploadSize = parseUploadSize(maxUploadSizeString);
+        }
+
+        ((UpdatableMultipartConfigElement) multipartConfigElement).setMaxFileSize(uploadSize);
+        this.configuration.setMaxFileUploadSize(maxUploadSizeString);
+
+        return true;
+    }
+
+    private Long parseUploadSize(String maxUploadSizeString) throws SizeFormatValidationException
+    {
+        String sizeUnit = maxUploadSizeString.replaceAll("[^A-Z]", "");
+        Long size = Long.parseLong(maxUploadSizeString.replaceAll("[^0-9]", ""));
+
+        int pow;
+        switch (sizeUnit)
+        {
+            case "B":
+                pow = 0; break;
+            case "KB": 
+                pow = 1; break;
+            case "MB":
+                pow = 2; break;
+            case "GB":
+                pow = 3; break;
```

Comment:
Cosmetic: `break;` statements should be on new line.

Reply:
done